src/journal/src/mem/stream.rs

```
@@ -43,10 +43,10 @@ impl<T: Timestamp> Stream for MemStream<T> {
     type EventStream = EventStream<Self::Timestamp>;
     type Timestamp = T;
 
-    async fn read_events(&self, ts: Self::Timestamp) -> Self::EventStream {
+    async fn read_events(&self, ts: Self::Timestamp) -> Result<Self::EventStream> {
```

read_events remotely can fail

---

You can return a error stream instead. Maybe something like `stream::once(future::err(...))`.

---

`stream::once` gives a `futures::stream::Once` which is incompatible with `journal::grpc::stream::EventStream`.

So we meet a similar issue as `Streaming<...>` vs. `Self::EventStream` which requires adapting the error stream type to `EventStream`.

---

Oh, that's just an example. What I mean is that we can return some stream that simply yields an error immediately.

---

I try to let EventStream becomes an enum that can be either a current EventStream or a wrapper of error to be yield when calling `poll_next`, but match on a pinning enum is quite tricky :(

I didn't find a good way to implement this "either" logic.